### PR TITLE
Add min amount into atomic exchange (v2-dev)

### PR DESCRIPTION
**Pull request by LeifuChen**
## Description
In `Menkent` release, atomic exchange is adding one more value `minAmount` for slippage control.

## Related issue
#672

## How Has This Been Tested?
1. Make a transaction sEUR => sUSD on L2. It triggers `exchangeWithTracking` method.
https://kovan-optimistic.etherscan.io/tx/0xe9264f2935addb93a63b73fc2c5a10df33f586ef9d3fac54f25a26028f5171f8
2. Make a transaction sEUR => sUSD on L1. It triggers `exchangeAtomically` method with the new parameter `minAmount`.
https://kovan.etherscan.io/tx/0x28a3468e6a1be67000217dd029ca2e9eb564eebb47094644f7d7a335a53b6c40

## Screenshots (if appropriate):
image

### CODE REVIEW

**Review by JChiaramonte7 — DISMISSED** [review]
LGTM but would be good to get a pass from the frontend devs either @koredefashokun or @Tburm

File: queries/synths/useBaseFeeRateQuery.ts [hunk 1]
```diff
@@ -7,22 +7,33 @@ import { CurrencyKey } from 'constants/currency';
...
-			const baseFeeRate = await SystemSettings.exchangeFeeRate(
-				ethers.utils.formatBytes32String(currencyKey as string)
+			const sourceCurrencyFeeRate = await SystemSettings.exchangeFeeRate(
+				ethers.utils.formatBytes32String(sourceCurrencyKey as string)
 			);
-			return baseFeeRate ? baseFeeRate : null;
+
+			const destinationCurrencyFeeRate = await SystemSettings.exchangeFeeRate(
+				ethers.utils.formatBytes32String(destinationCurrencyKey as string)
+			);
```
Comment (koredefashokun) [inline queries/synths/useBaseFeeRateQuery.ts:28]: Maybe we can wrap this in a `Promise.all`

File: queries/synths/useBaseFeeRateQuery.ts [hunk 2]
```diff
@@ -7,22 +7,33 @@ import { CurrencyKey } from 'constants/currency';
...
+
+			return sourceCurrencyFeeRate || destinationCurrencyFeeRate
+				? sourceCurrencyFeeRate.add(destinationCurrencyFeeRate)
+				: null;
```
Comment (koredefashokun) [inline queries/synths/useBaseFeeRateQuery.ts:32]: Do we want to check if both values exist or only one? I'm worried that there might be a case where one of `sourceCurrencyFeeRate` and `destinationCurrencyFeeRate` may be falsy, but the code will still attempt to run line 31, causing an error.

File: sections/exchange/hooks/useExchange.tsx [hunk 3]
```diff
@@ -765,11 +780,16 @@ const useExchange = ({
...
+
 			const isAtomic =
-				destinationCurrencyKey === 'sBTC' ||
-				destinationCurrencyKey === 'sETH' ||
-				destinationCurrencyKey === 'sEUR';
+				!isL2 &&
+				(destinationCurrencyKey === 'sBTC' ||
+					destinationCurrencyKey === 'sETH' ||
+					destinationCurrencyKey === 'sEUR');
+			console.log(`submit`, isAtomic);
```
Comment (koredefashokun) [inline sections/exchange/hooks/useExchange.tsx:792]: I think you forgot this `console.log`

**Review by koredefashokun — COMMENTED** [review]
Looks good. Left a few comments.

File: queries/synths/useBaseFeeRateQuery.ts (hunk 1 above)
Comment (LeifuChen) [reply queries/synths/useBaseFeeRateQuery.ts]: Yes. Thanks.

File: queries/synths/useBaseFeeRateQuery.ts (hunk 2 above)
Comment (LeifuChen) [reply queries/synths/useBaseFeeRateQuery.ts]: Thanks. The change has been made.

**Comment by LeifuChen** [comment]
@koredefashokun Thanks for the review. The changes have been made as requested.

File: queries/synths/useBaseFeeRateQuery.ts [hunk 4]
```diff
@@ -6,23 +6,36 @@ import { ethers } from 'ethers';
...
-			);
-			return baseFeeRate ? baseFeeRate : null;
+			const [sourceCurrencyFeeRate, destinationCurrencyFeeRate] = (await Promise.all([
+				SystemSettings.exchangeFeeRate(
+					ethers.utils.formatBytes32String(sourceCurrencyKey as string)
+				),
+				SystemSettings.exchangeFeeRate(
+					ethers.utils.formatBytes32String(destinationCurrencyKey as string)
+				),
+			])) as [Wei, Wei];
```
Comment (koredefashokun) [inline queries/synths/useBaseFeeRateQuery.ts:30]: Just a nitpick, but I think this might not be `Wei`, but rather `ethers.BigNumber`.

**Review by koredefashokun — APPROVED** [review]
Looks good. I left a small comment relating to the type of the result from contract calls.

File: queries/synths/useBaseFeeRateQuery.ts (hunk 4 above)
Comment (LeifuChen) [reply queries/synths/useBaseFeeRateQuery.ts:30]: Thanks. I will make the change.

Comment (LeifuChen) [reply queries/synths/useBaseFeeRateQuery.ts:30]: @koredefashokun I just took another look into the `TradeSummaryCard`, where we passed the `baseFeeRate`. Do you think we should change everything here?
image

Comment (LeifuChen) [reply queries/synths/useBaseFeeRateQuery.ts:34]: @koredefashokun Change has been made. Thanks.

**Review by koredefashokun — APPROVED** [review]

**Pull request merged**